pkg/config/config.go

```diff
@@ -981,6 +981,21 @@ func getMultipleEndpointsWithConfig(config Config) (map[string][]string, error)
 	return keysPerDomain, nil
 }
 
+// ComputeAPIDomain returns a domain to be used for API validation
+func ComputeAPIDomain(domain string) (string, error) {
+	apiDomain := ""
+	parsedDomain, err := url.Parse(domain)
+	if err != nil {
+		return "", fmt.Errorf("Could not parse %s as a URL", domain)
```

Comment:
nit: error strings should start with a lowercase letter, log messages with a capital letter, so here:

```suggestion
		return "", fmt.Errorf("could not parse %s as a URL", domain)
```